hal/platform/atmelsam/i2c_common/i2c_types.h

```diff
@@ -0,0 +1,31 @@
+//
+// Created by J.A. Korten on 03/11/2023.
+//
+
+#ifndef ATMELSAMD21_I2C_TYPES_H
+#define ATMELSAMD21_I2C_TYPES_H
+
+// See SAMD21 datasheet table 28.10.7
+
+typedef enum {
```

Comment:
Why not put this enum in i2c_platform_specific.h since these busconditions are very platform dependent. An STM32 peripheral might not or might have other busconditions/naming.